src/tokenizer/mod.rs

```diff
@@ -218,6 +220,73 @@ mod test {
         assert_eq!(&tokens[4], "やばい");
     }
 
+    #[test]
+    fn test_ngram_tokenizer() {
+        use super::{LowerCaser, NgramTokenizer};
+        use tokenizer::tokenizer::Tokenizer;
+
+        let tokenizer_manager = TokenizerManager::default();
+        tokenizer_manager.register("ngram12", NgramTokenizer::new(1, 2, false));
+        tokenizer_manager.register(
+            "ngram3",
+            NgramTokenizer::new(3, 3, false).filter(LowerCaser),
+        );
+        tokenizer_manager.register(
+            "edgegram5",
+            NgramTokenizer::new(2, 5, true).filter(LowerCaser),
+        );
+
+        let en_tokenizer = tokenizer_manager.get("ngram12").unwrap();
+        let mut tokens: Vec<String> = vec![];
+        {
+            let mut add_token = |token: &Token| {
+                tokens.push(token.text.clone());
+            };
+            en_tokenizer.token_stream("hello").process(&mut add_token);
+        }
+
+        assert_eq!(tokens.len(), 8);
+        assert_eq!(&tokens[0], "h");
+        assert_eq!(&tokens[1], "he");
+        assert_eq!(&tokens[2], "e");
+        assert_eq!(&tokens[3], "el");
+        assert_eq!(&tokens[4], "l");
+        assert_eq!(&tokens[5], "ll");
+        assert_eq!(&tokens[6], "l");
+        assert_eq!(&tokens[7], "lo");
```

Comment:
`assert` on positions

Reply:
I'm going to derive `Clone` and `Debug` for `Token` to make this a bit easier.

Reply:
OMG, clutch. I added token tests to the other tokenizers. I get it now. :)